### PR TITLE
Clean up properties, fix #7

### LINKED ISSUE #7: ClassName is not considered a property
For Instances, ClassName is not considered a property in the internal `properties` table. Although the only scenario this has mattered in is #5 (which doesn't matter because ClassName never changes anyway), this could become an issue in the future. Best to implement readonly properties and include ClassName now.

### PR DESCRIPTION
**Pull request by Kampfkarren**
- Adds "Instance.InternalProperty" and "Instance.InternalPropertyReadonly" generators to clean up properties (the Values could use cleaning up with this).
- Uses this to make ClassName a readonly property, fixing #7.

### CODE REVIEW

File: lib/Instance.lua [hunk 1]
```diff
@@ -155,7 +171,7 @@ end
 function Instance:IsA(className)
 	-- TODO: Hierarchy stuff
 
-	return self.ClassName == className
+	return self._internal.className == className
```
Comment (LPGhatguy) [inline lib/Instance.lua]: This line should still work as it was written, right?

File: lib/Instance.lua [hunk 2]
```diff
@@ -3,8 +3,33 @@ local Signal = import("./Signal")
...
 
+function Instance.InternalProperty(key)
+    return {
+        get = function(self)
+            return self._internal[key]
+        end,
+        set = function(self, _, value)
+            self._internal[key] = value
+        end,
+    }
```
Comment (LPGhatguy) [inline lib/Instance.lua]: Indentation! It looks like these blocks are indented with spaces.

I really want to build a check for all of our open source projects to catch this. For now, I just habitually highlight blocks in the GitHub review window to try to catch this.

File: lib/Instance.lua (hunk 1 above)
Comment (Kampfkarren) [reply lib/Instance.lua]: It should, I think I was just annoyed at some other stuff not working. I'll change it.

File: lib/Instance.lua (hunk 2 above)
Comment (Kampfkarren) [reply lib/Instance.lua]: luacheck sometimes complains about it, and sometimes doesn't.

Comment (LPGhatguy) [reply lib/Instance.lua]: I think Luacheck will sometimes only catch lines that have _both_ tabs and spaces, but not one or the other.

My check will basically just regex for `^\s- ` and error if any matches are found.

**Comment by LPGhatguy** [comment]
Nice job!

**Pull request merged**